### ISSUE TITLE
EGA 1440p shader is selected at 1080p resolution

### ISSUE THREAD

**Issue by weirddan455** [dosbox-staging/dosbox-staging#3334]
### Are you using the latest Dosbox-Staging Version?

- [X] I have checked releases and am using the latest release.

### Different version than latest?

Git Main

### What Operating System are you using?

Linux x86_64

### If Other OS, please describe

Arch Linux

### Have you checked that no other similar issue already exists?

- [X] I have searched and not found similar issues.

### A clear and concise description of what the bug is.

The shader `crt/ega-1440p` gets selected when viewed at a real 1080p resolution.  Unsure if this is intended.  I would assume `crt/ega-1080p` would be the correct shader to use.  Tested with Space Quest III.

As seen from the log below, `crt/ega-1080p` gets selected when the game is launched in windowed mode.  When I switch to fullscreen on my 1080p monitor, the shader switches to `crt/ega-1440p`

### Steps to reproduce the behaviour.

Explain how to reproduce
1.   Launch Space Quest III (possibly also reproducible with other EGA games).
2.  Alt + Tab to go fullscreen on a 1080p monitor
3.  Observe from the log that the 1440p shader gets selected

### Your configuration

```ini
Default config
```

### Provide a Log

```text
2024-01-18 10:51:01.549 | RENDER: Auto-switched to shader 'crt/vga-1080p'
2024-01-18 10:51:01.549 | SDL: Using standard SDI (auto) display refresh rate of 144 Hz
2024-01-18 10:51:01.555 | DISPLAY: VGA 720x400 16-colour text mode 03h at 70.087 Hz VFR, scaled to 1067x800 pixels with 1:1.35 (20:27) pixel aspect ratio
2024-01-18 10:51:01.555 | MOUNT: Path 'build/release/resources/drives/y' found
2024-01-18 10:51:01.556 | MOUNT: Path '/home/daniel/dos' found
2024-01-18 10:51:06.650 | RENDER: Auto-switched to shader 'crt/ega-1080p'
2024-01-18 10:51:06.650 | VGA: Forcing single scanning of double-scanned VGA video modes
2024-01-18 10:51:06.657 | DISPLAY: EGA 320x200 16-colour graphics mode 0Dh at 70.086 Hz VFR, scaled to 1067x800 pixels with 1:1.2 (5:6) pixel aspect ratio
2024-01-18 10:51:07.142 | MT32: LCD-Message:   INSERT BUCKAZOID  
2024-01-18 10:51:21.363 | MT32: LCD-Message: * SPACE QUEST III * 
2024-01-18 10:51:37.351 | SDL: Using VRR-adjusted (auto) display refresh rate of 141 Hz
2024-01-18 10:51:37.354 | RENDER: Auto-switched to shader 'crt/ega-1440p'
2024-01-18 10:51:37.356 | DISPLAY: EGA 320x200 16-colour graphics mode 0Dh at 70.086 Hz CFR, scaled to 1333x1000 pixels with 1:1.2 (5:6) pixel aspect ratio
```

### Code of Conduct & Contributing Guidelines

- [X] Yes, I agree.

**Comment by johnnovak** [comment]
This is expected @weirddan455 and normal. Think of those shader filenames as implementation detail. If you check out `shader_manager.cpp` you'll see the cutoff points for the resolution variants are defined in "pixels per native scanline". I could put those numbers into the shader names (e.g., `ega-5.glsl`), but I found the 720p, 1080p, etc. quallifiers that roughly match more informative.

If you restrict the physical image dimension to always smaller than fullscreen with `viewport = 89%` for example, the 1440p, 4k, etc. names will match your monitor's native res quite accurately. The shaders were tweaked for such more authentic image dimensions. Without this restriction, there is some bleeding into the next higher res variant in fullscreen.

For the record, EGA in 100% fullscreen is wholly inauthentic as the resulting physical image size on a 24" flat screen would correspond to a 21" CRT monitor. EGA monitors did not go above 13-14", so that's wholly inauthentic. Just explaining why I tweaked the names and the shaders themselves using authentic image sizes -- of course you or anyone can use them however they like it 😄 

Ultimately, the 20+ distinct shaders will go away and you'll only be able to override Hercules, EGA, VGA, etc. manually, but never the resolution. So all this detail will be completely hidden. All these shaders will be collapsed into a single one, and the switching will become a simple shader preset switching.
